features/dd-sdk-android-session-replay-material/src/test/kotlin/com/datadog/android/sessionreplay/material/ChipWireframeMapperTest.kt

```diff
@@ -0,0 +1,208 @@
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(value = ForgeConfigurator::class)
+class ChipWireframeMapperTest {
+
+    private lateinit var testedChipWireframeMapper: ChipWireframeMapper
+
+    @Mock
+    lateinit var mockChipView: Chip
+
+    @Mock
+    lateinit var mockViewBoundsResolver: ViewBoundsResolver
+
+    @Mock
+    lateinit var mockAsyncJobStatusCallback: AsyncJobStatusCallback
+
+    @Mock
+    lateinit var mockInternalLogger: InternalLogger
+
+    @Mock
+    lateinit var mockViewIdentifierResolver: ViewIdentifierResolver
+
+    @Mock
+    lateinit var mockColorStringFormatter: ColorStringFormatter
+
+    @Mock
+    lateinit var mockDrawableToColorMapper: DrawableToColorMapper
+
+    @Mock
+    lateinit var mockChipDrawable: Drawable
+
+    @LongForgery
+    var fakeViewId: Long = 0L
+
+    @StringForgery
+    var fakeText: String = ""
+
+    @Forgery
+    lateinit var fakeMappingContext: MappingContext
+
+    @Forgery
+    lateinit var fakeGlobalBounds: GlobalBounds
+
+    lateinit var fakeDrawableBounds: Rect
+
+    @IntForgery
+    private var fakeDrawableHeight: Int = 0
+
+    @IntForgery
+    private var fakeDrawableWidth: Int = 0
+
+    @Mock
+    lateinit var mockLayout: Layout
+
+    @FloatForgery(0f, 255f)
+    var fakeFontSize: Float = 0f
+
+    @IntForgery(min = 0, max = 0xffffff)
+    var fakeTextColor: Int = 0
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
+        fakeDrawableBounds = Rect(
+            forge.aSmallInt(),
+            forge.aSmallInt(),
+            forge.aSmallInt(),
+            forge.aSmallInt()
+        )
+        mockChipView = mockChip()
+        whenever(
+            mockViewBoundsResolver.resolveViewGlobalBounds(
+                mockChipView,
+                fakeMappingContext.systemInformation.screenDensity
+            )
+        ).thenReturn(fakeGlobalBounds)
+        whenever(
+            mockViewIdentifierResolver.resolveViewId(
+                mockChipView
+            )
+        ).thenReturn(fakeViewId)
+        testedChipWireframeMapper = ChipWireframeMapper(
+            viewIdentifierResolver = mockViewIdentifierResolver,
+            colorStringFormatter = mockColorStringFormatter,
+            viewBoundsResolver = mockViewBoundsResolver,
+            drawableToColorMapper = mockDrawableToColorMapper
+        )
+    }
+
+    @Test
+    fun `M resolves card view wireframe W map`() {
+        // When
+        testedChipWireframeMapper.map(
+            mockChipView,
+            fakeMappingContext,
+            mockAsyncJobStatusCallback,
+            mockInternalLogger
+        )
+
+        // Then
+        val density = fakeMappingContext.systemInformation.screenDensity
+
+        verify(
+            fakeMappingContext.imageWireframeHelper,
+            times(1)
```

Comment:
you can remove this one, it is by default like that

Reply:
updated